packages/accounts-controller/src/AccountsController.ts

```diff
@@ -306,7 +377,20 @@ export class AccountsController extends BaseController<
         ...account,
         metadata: { ...account.metadata, name: accountName },
       };
-      currentState.internalAccounts.accounts[accountId] = internalAccount;
+      // deep clone of old state to get around Type instantiation is excessively deep and possibly infinite.
+      const oldState = JSON.parse(JSON.stringify(currentState));
```

Comment:
I'm surprised we don't have a helper for this already? Looks like we could this maybe: https://developer.mozilla.org/en-US/docs/Web/API/structuredClone

Reply:
Also, we could add our own `deepClone` function that keeps type information like:

```ts
function deepClone<O>(obj: O): O {
  return JSON.parse(JSON.stringify(obj)) as O;
}
```

So that, at least `oldState` would still be typed as `AccountsControllerState`, WDYT?

Reply:
Using structuredClone and deepClone from lodash yielded the same error.